jenkins/freestyle/delete-old-workers.groovy

```diff
@@ -0,0 +1,19 @@
+import hudson.model.*
+import hudson.node_monitors.*
+import hudson.slaves.*
+import java.util.concurrent.*
+
+import hudson.model.*;
+
+jenkins = Hudson.instance
+
+for (slave in jenkins.computers) {
+  def computer = slave.name
+    if (slave.name != "master" & slave.isOffline() & slave.offlineCause != null) {
+        offCause = slave.getOfflineCause().toString()
+        if (offCause.contains("Time out for last 5 try")) {
+          println(slave.name + " " + slave.offlineCause + " = " + slave.isOffline())
```

Comment:
This printout seems a bit confusing to me. E.g. it'll say something like:

> jenkins-worker (i-4288c5c1) Time out for last 5 try = true

But that message not exactly correct. Time out for the last 5 try is true because you're in the if clause, but the " = true" actually gets put there after recalculating whether the slave is **still** offline by another call to slave.isOffline().